sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/QueueSenderAsyncClient.java

```diff
@@ -0,0 +1,353 @@
+import com.azure.core.amqp.exception.AmqpErrorCondition;
+import com.azure.core.amqp.exception.AmqpException;
+
+import com.azure.core.amqp.implementation.AmqpSendLink;
+import com.azure.core.amqp.implementation.ErrorContextProvider;
+
+import static com.azure.core.amqp.implementation.RetryUtil.getRetryPolicy;
+import static com.azure.core.amqp.implementation.RetryUtil.withRetry;
+
+import com.azure.core.amqp.implementation.MessageSerializer;
+import com.azure.core.amqp.implementation.TracerProvider;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.util.Context;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.tracing.ProcessKind;
+import com.azure.messaging.servicebus.implementation.CreateBatchOptions;
+import com.azure.messaging.servicebus.implementation.ServiceBusConnectionProcessor;
+import org.apache.qpid.proton.amqp.messaging.MessageAnnotations;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Signal;
+
+import java.io.Closeable;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BiConsumer;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+
+import static com.azure.core.util.tracing.Tracer.ENTITY_PATH_KEY;
+import static com.azure.core.util.tracing.Tracer.HOST_NAME_KEY;
+import static com.azure.core.util.tracing.Tracer.SPAN_CONTEXT_KEY;
+
+/**
+ * The client to send messages to Queue.
+ */
+@ServiceClient(builder = QueueClientBuilder.class, isAsync = true)
+public final class QueueSenderAsyncClient implements Closeable {
+
+    private final ClientLogger logger = new ClientLogger(QueueSenderAsyncClient.class);
+    private final AtomicBoolean isDisposed = new AtomicBoolean();
+    private final TracerProvider tracerProvider;
+    private final MessageSerializer messageSerializer;
+    private final AmqpRetryOptions retryOptions;
+    private final AmqpRetryPolicy retryPolicy;
+    private final String queueName;
+    private final ServiceBusConnectionProcessor connectionProcessor;
+
+    /**
+     * The default maximum allowable size, in bytes, for a batch to be sent.
+     */
+    public static final int MAX_MESSAGE_LENGTH_BYTES = 256 * 1024;
+
+    /**
+     * Creates a new instance of this {@link QueueSenderAsyncClient} that sends messages to
+     */
+    QueueSenderAsyncClient(String queueName, ServiceBusConnectionProcessor connectionProcessor,
+                           AmqpRetryOptions retryOptions, TracerProvider tracerProvider,
+                           MessageSerializer messageSerializer) {
```

Comment:
What are some examples of message serializers? I'm guessing we're all going to have JSON, strings. Do you have any Java specific serializers?

Reply:
Currently, this serializer will support converting AMQP message in Java SDK representation `Message` . Making sure the property from AMQP message are translated properly. As of now Binary and json will be supported. We have to bring parity with track 1.